NBitcoin/TaprootAddress.cs

```diff
@@ -0,0 +1,119 @@
+﻿#if HAS_SPAN
+#nullable enable
+using NBitcoin.DataEncoders;
+using NBitcoin.Secp256k1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NBitcoin
+{
+	public class TaprootAddress : BitcoinAddress, IBech32Data, IPubkeyHashUsable
+	{
+		public TaprootAddress(string bech32, Network expectedNetwork)
+				: base(Validate(bech32, expectedNetwork), expectedNetwork)
+		{
+			if (expectedNetwork.GetBech32Encoder(Bech32Type.TAPROOT_ADDRESS, false) is Bech32Encoder encoder)
+			{
+				var decoded = encoder.Decode(bech32, out _);
+				if (ECXOnlyPubKey.TryCreate(decoded, out var k))
+					_PubKey = new PubKey(k);
```

Comment:
Are you referring to this code with reusing the pukey?

Reply:
Yes. As you know a pubkey is just a point  (x, y) in the elliptic curve and given it is symmetric there are two point for each x coordinate (higher and lower). Before taproot we had that a public key was "compressed"  meaning that it was a 33 bytes pubkey (32 bytes corresponding to the X coordinate plus and additional byte to indicate which of the two points to use).

In Taproot we have a 32 bytes public key and the extras bit was just dropped, that's the `XOnlyPubKey`. Nicolas doubt makes sense to me because I don't know what the best approach is neither.

Reply:
Ah I see what you mean, intuitively I would say we should create a new pubkey type.

The options I see are:
-  new pubkey it can perhaps inherit from the pubkey and check the array limits to 32 bytes (then its reusable).
-  new type all together (no inheritance) this mean les compatibility with other components.
- dummy last byte in the array (but this is tricky as we cant find a dummy byte that is invalid).

Intuitively I think inheritance is the easiest option.